### ISSUE TITLE
ConnectionDetails Typescript interface

### ISSUE THREAD

**Issue by saberistic** [amqp/rhea#183]
Looking at the `reconnect` example, Shouldn't `options` and `connect` be optional in 
```ts
export interface ConnectionDetails {
  /**
   * @property {string} host The host name. Default value: `"localhost"`.
   */
  host: string;
  /**
   * @property {number} port The port number. Default value: `5672`.
   */
  port: number;
  /**
   * @property {any} options The options object provided to connection_details().
   */
  options: any;
  /**
   * @property {Function} connect The `connect` function of `"net"` or `"tls"` module.
   */
  connect: Function;
}
```
https://github.com/amqp/rhea/blob/master/typings/connection.d.ts#L41

**Comment by theonlystylar** [comment]
Agreed, I need to keep properties connect and options undefined.  I'm setting up failover with my ActiveMQ message broker (see https://github.com/amqp/rhea/blob/master/examples/reconnect/client.js#L43) and I need to leave options & connect undefined.

My current workaround was to override the typings (see these instructions: https://davidgomes.com/maintaining-overridden-typescript-definitions-for-dependencies/).

**Comment by theonlystylar** [comment]
I created a pull request: https://github.com/amqp/rhea/pull/238 to make the options and connect properties optional.
